packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -260,6 +261,34 @@ describe('KeyringController', () => {
     });
   });
 
+  describe('behavior when keyring metadata length mismatch', () => {
+    it('should throw an error if the keyring metadata length mismatch', async () => {
+      let vaultWithOneKeyring;
+      await withController(async ({ controller }) => {
+        vaultWithOneKeyring = controller.state.vault;
+      });
+
+      await withController(
+        {
+          skipVaultCreation: true,
+          state: {
+            vault: vaultWithOneKeyring,
```

Comment:
hmm.. I'm not sure that we are testing what we expect to be testing here, because the vault encryptor is mocked during tests 

probably it works anyway because the controller will be restored with the last object that was passed to the encryptor, which has one keyring only